src/admin/units/deleteUnitsDialog.component.test.tsx

```diff
@@ -0,0 +1,88 @@
+import { RenderResult } from '@testing-library/react';
+import { Unit } from '../../app.types';
+import { DeleteUnitProps } from './deleteUnitsDialog.component';
+import DeleteUnitDialog from './deleteUnitsDialog.component';
+import { renderComponentWithRouterProvider } from '../../testUtils';
+import userEvent, { UserEvent } from '@testing-library/user-event';
+import { screen, waitFor } from '@testing-library/react';
+import handleIMS_APIError from '../../handleIMS_APIError';
+
+vi.mock('../../handleIMS_APIError');
+
+describe('delete Unit dialog', () => {
+  let props: DeleteUnitProps;
+  let user: UserEvent;
+  const onClose = vi.fn();
+  let unit: Unit;
+  const createView = (): RenderResult => {
+    return renderComponentWithRouterProvider(<DeleteUnitDialog {...props} />);
+  };
+
+  beforeEach(() => {
+    (unit = {
+      id: '1',
+      value: 'test',
+      created_time: '2024-01-01T12:00:00.000+00:00',
+      modified_time: '2024-01-02T13:10:10.000+00:00',
+    }),
+      (props = {
+        open: true,
+        onClose: onClose,
+        unit: unit,
+      });
+    user = userEvent.setup();
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it('displays warning message when session data is not loaded', async () => {
+    props = {
+      ...props,
+      unit: undefined,
+    };
+    createView();
+    const continueButton = screen.getByRole('button', { name: 'Continue' });
+    await user.click(continueButton);
+    const helperTexts = screen.getByText(
+      'No data provided. Please refresh and try again'
+    );
+    expect(helperTexts).toBeInTheDocument();
+    expect(onClose).not.toHaveBeenCalled();
+  });
+
+  it('calls handleDeleteSession when continue button is clicked with a valid session name', async () => {
```

Comment:
I don't think this name makes sense. I think there are a few others if you search for `valid session name` could you change those as well please.